lib/ip_blocklist.js

```diff
@@ -0,0 +1,165 @@
+    self.filePath = filePath
+    self.fileName = path.basename(filePath)
+
+    var startTime = Date.now()
+    return statFile(filePath)
+      .then(function (fileStats) {
+        self.fileLastModified = fileStats.mtime
+        self.fileSize = fileStats.size
+        return readFile(filePath, 'utf8')
+      })
+      .then(function (data) {
+        return parse(data)
+      })
+      .then(function (rows) {
+        rows.forEach(function (row) {
+          var parsedData = parseRow(row[0])
+
+          if (parsedData) {
+            if (!self.ipsByPrefixLength[parsedData.prefixLength]) {
+              self.prefixLengths.push(parsedData.prefixLength)
+              self.ipsByPrefixLength[parsedData.prefixLength] = {}
+            }
+            self.ipsByPrefixLength[parsedData.prefixLength][parsedData.baseIp] = true
+          }
+        })
+
+        var endTime = Date.now()
+        log.info({
+          op: 'fxa.customs.blocklist.load',
+          fileName: self.fileName,
+          fileSize: self.fileSize,
+          loadedIn: (endTime - startTime)
+        })
+      })
+  }
+
+  IPBlocklist.prototype.clear = function () {
+    this.prefixLengths = []
+    this.ipsByPrefixLength = {}
+  }
+
+  IPBlocklist.prototype.contains = function (ipAddress) {
+    var self = this
+
+    var startTime = Date.now()
+    var endTime
+
+    log.info({
+      op: 'fxa.customs.blocklist.check',
+      ip: ipAddress,
+      fileName: self.fileName,
+      fileSize: self.fileSize
+    })
+
+    try {
+      return self.prefixLengths.some(function (prefixLength) {
+        var result = getBaseIp(ipAddress, prefixLength) in self.ipsByPrefixLength[prefixLength]
+
+        // If `hit` log metrics on time taken
+        if (result) {
+          endTime = Date.now()
+          log.info({op: 'fxa.customs.blocklist.hit', ip: ipAddress, fileName: self.fileName, fileSize: self.fileSize,
+            foundIn: (endTime - startTime)})
+        }
+
+        return result
+      })
+    } catch (err) {
+      // On any error, fail closed and reject request. These errors could be
+      // malformed ip address, etc
+      log.error({op: 'fxa.customs.blocklist.error', fileName: self.fileName, fileSize: self.fileSize, err: err})
+      return true
+    }
+  }
+
+  IPBlocklist.prototype.pollForUpdates = function () {
+    this.stopPolling()
+    this.pollInterval = setInterval(this.refresh.bind(this), config.ipBlocklist.updatePollInterval * 1000)
+  }
+
+  IPBlocklist.prototype.stopPolling = function () {
+    clearInterval(this.pollInterval)
+  }
+
+  IPBlocklist.prototype.refresh = function () {
+    var self = this
+
+    log.trace({op: 'fxa.customs.blocklist.refreshList', fileName: self.fileName, fileSize: self.fileSize})
+
+    return statFile(self.filePath)
+      .then(function (fileStats) {
+        var mtime = fileStats.mtime
+        if (mtime > self.fileLastModified) {
+          self.fileLastModified = mtime
+          return self.load(self.filePath)
```

Comment:
I think from doing https://github.com/mozilla/fxa-customs-server/pull/117#discussion_r79723749, it will ensure a clean list is loaded. 

> Another concern is, how long does the reload typically take?

Reloading on a 1mb file typically takes 300ms on my computer and will only occur if a change in the file is detected. A scenario I could see blocking event loop significantly is if all blocklist files were updated, then they could all try to reload themselves at the same time. 

Not sure if that is best handled in here or in the updating script?